geomesa-fs/geomesa-fs-datastore/src/main/scala/org/locationtech/geomesa/fs/FileSystemFeatureIterator.scala

```diff
@@ -0,0 +1,151 @@
+/***********************************************************************
+ * Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+
+package org.locationtech.geomesa.fs
+
+import java.util
+import java.util.concurrent._
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.fs.FileSystem
+import org.geotools.data.Query
+import org.locationtech.geomesa.fs.storage.api.{FileSystemStorage, Partition, PartitionScheme}
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.util.control.NonFatal
+
+class FileSystemFeatureIterator(fs: FileSystem,
```

Comment:
Looks like this class could be replaced with 

FsQueryPlanning.getPartitionsForQuery(storage, sft, q).map{ new ThreadedReader(storage, partitions, sft.getTypeName, q, readThreads) }